### PR TITLE
Web Google Auth

### PR DESCRIPTION
**Pull request by jakcharvat**
Implement the web portion of the plugin

### CODE REVIEW

**Review by PavelHolman — APPROVED** [review]

File: src/web.ts [hunk 1]
```diff
@@ -1,15 +1,150 @@
...
   }
-  signIn(): Promise<User> {
-    throw this.unimplemented('Sign in is not yet available on the web');
+
+  async loadScript(): Promise<void> {
+    if (!(window instanceof Window) || !(document instanceof Document)) {
+      throw new InitializationError('NoDocumentOrWindow');
+    }
+
+    const scriptId = 'gapi';
```
Comment (Kiwow) [inline src/web.ts:64]: Isn't this the same constant as `GAPI_SCRIPT_ID`?

File: src/web.ts [hunk 2]
```diff
@@ -1,15 +1,150 @@
 import { WebPlugin } from '@capacitor/core';
-
 import type { GoogleAuthPlugin, InitOptions, User } from './definitions';
 
+const GAPI_SCRIPT_ID = 'gapi' as const;
+const GAPI_SRC = 'https://apis.google.com/js/platform.js' as const;
```
Comment (Kiwow) [inline src/web.ts:5]: Pretty sure there's no need for `as const` here, the literal types get inferred for strings just fine.

Definitely not a war crime though, I can see a case being made for semantically marking these as "actual" constants.

File: src/web.ts [hunk 3]
```diff
@@ -1,15 +1,150 @@
...
+  async initialize(
+    _options: Partial<InitOptions> = {
+      clientId: '',
+      scopes: [],
+    },
+  ): Promise<void> {
+    const metaClientId = (
+      document.getElementsByName(
+        'google-signin-client_id',
+      )[0] as HTMLMetaElement
```
Comment (Kiwow) [inline src/web.ts]: This cast should be replaced by an `instanceof` check in my opinion.

Also, the way it is now, `metaClientId`'s type does not include `undefined`, which is incorrect.

File: src/web.ts [hunk 4]
```diff
@@ -1,15 +1,150 @@
...
+    script.async = true;
+    script.id = scriptId;
+
+    await new Promise<void>((resolve, reject) => {
+      script.onload = () => resolve();
+      script.onerror = (_event, _source, _lineno, _colno, error) =>
+        reject(error);
+      script.src = GAPI_SRC;
+      head.appendChild(script);
+    });
```
Comment (Kiwow) [inline src/web.ts:83]: Nice

**Review by Kiwow — APPROVED** [review]
Just a few nitpicks and altogether one thing that affects the plugin at runtime.

Otherwise LGTM 👍

File: src/web.ts (hunk 1 above)
Comment (jakcharvat) [reply src/web.ts:64]: yup

**Pull request merged**